src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -360,7 +360,7 @@ public static string DecodeString(IByteBuffer src, int readerIndex, int len, Enc
                 try
                 {
                     buffer.WriteBytes(src, readerIndex, len);
-                    ArraySegment<byte> ioBuf = src.GetIoBuffer();
+                    ArraySegment<byte> ioBuf = buffer.GetIoBuffer();
```

Comment:
I assume it means using composite buffers and ToString is broken in 0.4.7, right?

Reply:
Yes, found it in http branch, should be buffer to copy from.